android/src/com/mapswithme/maps/routing/NavigationController.java

```diff
@@ -270,4 +351,108 @@ public NavMenu getNavMenu()
   {
     return mNavMenu;
   }
+
+  private void toggleSearch()
+  {
+    final int animRes;
+    if (mIsSearchExpanded)
+    {
+      animRes = R.animator.show_zoom_out_alpha;
```

Comment:
Так можно писать, компилятор понимает что final определён в обеих ветках if-else. А перед ифом он только объявлен.

> On Sep 9, 2016, at 5:32 PM, Vladimir Byko-Ianko <notifications@github.com> wrote:
> 
> In android/src/com/mapswithme/maps/routing/NavigationController.java <https://github.com/mapsme/omim/pull/4246#discussion_r78190202>:
> 
> > @@ -270,4 +351,108 @@ public NavMenu getNavMenu()
> >    {
> >      return mNavMenu;
> >    }
> > +
> > +  private void toggleSearch()
> > +  {
> > +    final int animRes;
> > +    if (mIsSearchExpanded)
> > +    {
> > +      animRes = R.animator.show_zoom_out_alpha;
> Дима, поясни п-та. Ты определил final int animRes; а затем ее переопределил animRes = R.animator.show_zoom_out_alpha; 
> Нужно ли слово final в этом контексте?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/mapsme/omim/pull/4246/files/f104f4709dfa59a30e3d90cfa6d858d0b51ba4ec#r78190202>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ACKTRLPae9IfhAFUFl_xoJFz7Kefo4rHks5qoW4KgaJpZM4J4yII>.
>